packages/langium/src/grammar/internal-grammar-util.ts

```diff
@@ -50,7 +54,7 @@ function isDataTypeRuleInternal(rule: ast.ParserRule, visited: Set<ast.ParserRul
             return false;
         }
     }
-    return true;
+    return !!rule.definition;
```

Comment:
In general I prefer writing `Boolean(rule.definition` over double negation because it expresses more directly what it actually does.

Reply:
Changed